janusgraph-core/src/main/java/org/janusgraph/diskstorage/util/BufferUtil.java

```diff
@@ -57,6 +64,39 @@ public static StaticBuffer getIntBuffer(int[] ids) {
         return StaticArrayBuffer.of(arr);
     }
 
+    /**
+     * If string is not 8 bytes, then we write it as it is. Otherwise, we write a special
+     * marker at the end, so that when JanusGraph sees the buffer, it knows it should
+     * reinterpret the byte buffer as string buffer. See {@link IDManager#getKeyID(StaticBuffer)}
+     * for more details.
+     *
+     * @param s
+     * @return
+     */
+    public static StaticBuffer getStringBuffer(String s) {
```

Comment:
If we don't write a special marker at the end for a string with 8 bytes, `IDManager#getKeyID(StaticBuffer)` will break because it uses the length of the buffer to judge whether it's string-type or long-type.

Reply:
Got it. I think the next question could be related: https://github.com/JanusGraph/janusgraph/pull/3418#discussion_r1172041147

I'm just thinking that JanusGraph instance should probably have some runtime information about type of id to be used (String ID, Long ID, or any other sort of ids). In such case, we probably won't need to use that marker.
I just think some of the logic regarding ID could be abstracted away and during JanusGraph instance creation we could configure it to use one or another logic for ids via some Strategy patterns or else.

Btw. I suspect if we decide on the above solution (i.e. without special marker) then some refactoring might be needed, but I can potentially help on weekends with it if you need a hand on this.